### PR TITLE
[BUILD] Add pyproject.toml

### PR DESCRIPTION
**Pull request by AndreSlavescu**
## Summary

added pyproject.toml

## Testing Done

ran `pip install -e .` and it built successfully

- Hardware Type: RTX 3090
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+[build-system]
+requires = ["setuptools>=42", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "liger_kernel"
+version = "0.1.0"
```
Comment (ByronHsu) [inline pyproject.toml:7]: should be 0.1.1

**Review by ByronHsu — COMMENTED** [review]
can we try delete setup.py see if that works

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+[build-system]
+requires = ["setuptools>=42", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "liger_kernel"
+version = "0.1.1"
+description = "Efficient Triton kernels for LLM Training"
+authors = [
```
Comment (ByronHsu) [inline pyproject.toml:9]: we might not need this part since the contributor will grow

**Comment by ByronHsu** [comment]
lgtm if we bump to the new ver

**Comment by ByronHsu** [comment]
@AndreSlavescu let's get this in sometimes

**Review by ByronHsu — DISMISSED** [review]

**Review by ByronHsu — APPROVED** [review]

**Review by yundai424 — APPROVED** [review]

**Pull request merged**